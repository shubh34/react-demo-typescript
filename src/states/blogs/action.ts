import uuid from 'node-uuid';
import moment from 'moment';
import { IBlog } from "./types";

export const ADD_BLOG: string = "ADD_BLOG";
export const EDIT_BLOG: string = "EDIT_BLOG";
export const DELETE_BLOG: string = "DELETE_BLOG";
export const VIEW_BLOG: string = "VIEW_BLOG";

export const addBlog = (blog: IBlog) => {
    return {
        type: ADD_BLOG,
        blog: {
            ...blog,
            id: uuid.v1(),
            date: moment(),
        },
    }
}

export const deleteBlog = (blogId: string) => {
    return {
        type: DELETE_BLOG,
        blogId: blogId,
    }
}

export const ediBlog = (blog: IBlog) => {
    return {
        type: EDIT_BLOG,
        blog,
    }
}




