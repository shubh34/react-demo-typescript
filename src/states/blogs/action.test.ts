import moment from 'moment';
import { addBlog, deleteBlog, DELETE_BLOG, edit_blog } from './action';

describe('blogs action', () => {
    it('should create add blog action', () => {
        expect(addBlog({
            id: '2',
            title:'Test Blog',
            content: 'Content of blog',
            date: moment('22 Jan 2018'),
            categories: ['cat1', 'cat2'],
        })).toMatchSnapshot();
    })
    it('should create delete blog action', () => {
        expect(deleteBlog('2')).toEqual({
            type: DELETE_BLOG,
            blogId:'2'
        });
    });
    it('should create edit blog action', () => {
        expect(edit_blog({
            id: '2',
            title:'Test Blog',
            content: 'Content of blog',
            date: moment('22 Jan 2018'),
            categories: ['cat1', 'cat2'],
        })).toMatchSnapshot();
    });
});