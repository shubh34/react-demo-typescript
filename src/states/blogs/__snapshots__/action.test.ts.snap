// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blogs action should create add blog action 1`] = `
Object {
  "blog": Object {
    "categories": Array [
      "cat1",
      "cat2",
    ],
    "content": "Content of blog",
    "date": "2017-02-14T12:51:48.000Z",
    "id": "uuid",
    "title": "Test Blog",
  },
  "type": "ADD_BLOG",
}
`;

exports[`blogs action should create edit blog action 1`] = `
Object {
  "blog": Object {
    "categories": Array [
      "cat1",
      "cat2",
    ],
    "content": "Content of blog",
    "date": "2018-01-21T23:00:00.000Z",
    "id": "2",
    "title": "Test Blog",
  },
  "type": "EDIT_BLOG",
}
`;
